examples/typechecking/named_constructor.an

```diff
@@ -8,6 +8,6 @@ foo = hello_foo 42
 
 // args: --check --show-types
 // expected stdout:
-// Foo : (forall a b c. (a - b -> (Foo a b) can c))
-// foo : (Foo String (Int a))
```

Comment:
The removal of parenthesis here changes the meaning of the type since `Foo String Int a` would be a type with 3 arguments.

Can we keep the parenthesis so a type like `Foo (Int a) String` would still display correctly?